src/components/Slider/index.js

```diff
@@ -0,0 +1,106 @@
+import React, { useState, useEffect } from "react"
+import styles from "./Slider.module.scss"
+
+const Slider = () => {
+   const [slideIndex, setSlideIndex] = useState(1)
+   const [images, setImages] = useState([])
+   const [isLoading, setIsLoading] = useState(false)
+
+   const plusSlides = (n) => {
+      if (slideIndex === 5 && n === 1) {
+         setSlideIndex(1)
+         return
+      }
+      if (slideIndex === 1 && n === -1) {
+         setSlideIndex(5)
+         return
+      }
+      setSlideIndex((prevIndex) => prevIndex + n)
+   }
+
+   const currentSlide = (n) => {
+      setSlideIndex(n)
+   }
+
+   useEffect(() => {
+      let counter = 1
+      const refresh = setInterval(function () {
+         setSlideIndex(counter)
+         counter++
+         if (counter === 6) {
+            counter = 1
+         }
+      }, 3000)
+      return () => {
+         clearInterval(refresh)
+      }
+   }, [])
+
+   useEffect(() => {
+      setIsLoading(true)
+      fetch("http://localhost:5000/banners", { method: "GET" })
+         .then((response) => response.json())
```

Comment:
can you take the api logic to an external file. Make your component free from fetch/axios. Create an APIService.js that would contain the fetch logic. Import APIService in this component. Also dont hardcode url, import them from some constant file

Reply:
A new APIService.js file created for all the fetch logic and url is now being imported from .env file